src/state/YouTubeAppReducer.js

```diff
@@ -0,0 +1,14 @@
+const YouTubeAppReducer = (state, action) => {
+  switch (action.type) {
+    case 'search':
+      return { ...state, searchTerm: action.payload };
+    case 'select-video':
+      return { ...state, currentVideo: action.payload };
+    case 'toggle-theme':
+      return { ...state, darkTheme: !state.darkTheme };
+    default:
+      throw Error(`Unsupported state: ${state.action}`);
+  }
+};
+
+export default YouTubeAppReducer;
```

Comment:
This look good to me, you can move the Actions to a file where you only have the actions defined as constants, this way you can reuse them, for example
instead of
```jsx
dispatch({ type: 'search', payload: 'asdf' });
```
you can do:
```jsx
import { SEARCH } from '../path/to/actions';

dispatch({ type: SEARCH, payload: 'asdf' });
```

this is really useful because you don't have to look at the reducer code to know which actions you have, you can just look at the file that contains all the actions